### PR TITLE
Export of the sensi params choice

### PR DESCRIPTION
**Pull request by terriervik**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change?** *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: ra-optimisation/linear-rao/src/test/java/com/farao_community/farao/linear_rao/json/JsonLinearRaoResultTest.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        roundTripTest(raoResult, JsonRaoResult::write, JsonRaoResult::read, "/RaoResultDefaultSensi.json");
+    }
+
+    @Test
+    public void roundTripFallbackSensiParams() throws IOException {
+        LinearRaoResult resultExtension = new LinearRaoResult();
+        resultExtension.setSystematicSensitivityAnalysisParameters(true);
+        raoResult.addExtension(LinearRaoResult.class, resultExtension);
+        roundTripTest(raoResult, JsonRaoResult::write, JsonRaoResult::read, "/RaoResultFallbackSensi.json");
+    }*/
```
Comment (TheMaskedTurtle) [inline ra-optimisation/linear-rao/src/test/java/com/farao_community/farao/linear_rao/json/JsonLinearRaoResultTest.java:50]: :angry:

Comment (terriervik) [reply ra-optimisation/linear-rao/src/test/java/com/farao_community/farao/linear_rao/json/JsonLinearRaoResultTest.java:50]: It doesn't even compile, let me finish my work

File: ra-optimisation/linear-rao/src/main/java/com/farao_community/farao/linear_rao/LinearRaoResult.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
+/*
+ *
+ *  * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ *  * This Source Code Form is subject to the terms of the Mozilla Public
+ *  * License, v. 2.0. If a copy of the MPL was not distributed with this
+ *  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
```
Comment (baptsegu) [inline ra-optimisation/linear-rao/src/main/java/com/farao_community/farao/linear_rao/LinearRaoResult.java:7]: something strange with two much '*' in the header

File: ra-optimisation/linear-rao/src/main/java/com/farao_community/farao/linear_rao/LinearRaoResult.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    private SystematicSensitivityAnalysisParameters systematicSensitivityAnalysisParameters;
+
+    @Override
+    public String getName() {
+        return "LinearRaoResult";
+    }
+
+    public SystematicSensitivityAnalysisParameters getSystematicSensitivityAnalysisParameters() {
+        return systematicSensitivityAnalysisParameters;
+    }
```
Comment (baptsegu) [inline ra-optimisation/linear-rao/src/main/java/com/farao_community/farao/linear_rao/LinearRaoResult.java:36]: It's just about the wording, but I would have prefer SystematicSensitivityAnalysis**Status**.
Parameters make me think about input configuration parameters.

File: ra-optimisation/linear-rao/src/main/java/com/farao_community/farao/linear_rao/json/JsonLinearRaoResult.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
+/*
+ *
+ *  * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ *  * This Source Code Form is subject to the terms of the Mozilla Public
+ *  * License, v. 2.0. If a copy of the MPL was not distributed with this
+ *  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
```
Comment (baptsegu) [inline ra-optimisation/linear-rao/src/main/java/com/farao_community/farao/linear_rao/json/JsonLinearRaoResult.java:7]: same

File: ra-optimisation/linear-rao/src/main/java/com/farao_community/farao/linear_rao/LinearRaoResult.java (hunk 3 above)
Comment (baptsegu) [reply ra-optimisation/linear-rao/src/main/java/com/farao_community/farao/linear_rao/LinearRaoResult.java:36]: and maybe with three values ? DEFAULT, FALLBACK or FAILURE ? depending if the sensi fails (whatever the config), or if it works with default or fallback parameters.

File: ra-optimisation/linear-rao/src/test/java/com/farao_community/farao/linear_rao/json/JsonLinearRaoResultTest.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
+/*
+ *
+ *  * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ *  * This Source Code Form is subject to the terms of the Mozilla Public
+ *  * License, v. 2.0. If a copy of the MPL was not distributed with this
+ *  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
```
Comment (baptsegu) [inline ra-optimisation/linear-rao/src/test/java/com/farao_community/farao/linear_rao/json/JsonLinearRaoResultTest.java:7]: same

File: ra-optimisation/rao-api/src/main/java/com/farao_community/farao/rao_api/json/RaoResultDeserializer.java [hunk 6]
```diff
@@ -0,0 +1,92 @@
+/*
+ *
+ *  * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ *  * This Source Code Form is subject to the terms of the Mozilla Public
+ *  * License, v. 2.0. If a copy of the MPL was not distributed with this
+ *  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ */
```
Comment (baptsegu) [inline ra-optimisation/rao-api/src/main/java/com/farao_community/farao/rao_api/json/RaoResultDeserializer.java:8]: here too :cactus:

File: ra-optimisation/rao-api/src/main/java/com/farao_community/farao/rao_api/json/RaoResultJsonModule.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
+/*
+ *
+ *  * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ *  * This Source Code Form is subject to the terms of the Mozilla Public
+ *  * License, v. 2.0. If a copy of the MPL was not distributed with this
+ *  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ */
```
Comment (baptsegu) [inline ra-optimisation/rao-api/src/main/java/com/farao_community/farao/rao_api/json/RaoResultJsonModule.java:8]: :stars:

File: ra-optimisation/rao-api/src/main/java/com/farao_community/farao/rao_api/json/RaoResultSerializer.java [hunk 8]
```diff
@@ -0,0 +1,43 @@
+/*
+ *
+ *  * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ *  * This Source Code Form is subject to the terms of the Mozilla Public
+ *  * License, v. 2.0. If a copy of the MPL was not distributed with this
+ *  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ */
```
Comment (baptsegu) [inline ra-optimisation/rao-api/src/main/java/com/farao_community/farao/rao_api/json/RaoResultSerializer.java:8]: :star:

File: ra-optimisation/linear-rao/src/main/java/com/farao_community/farao/linear_rao/LinearRao.java [hunk 9]
```diff
@@ -85,7 +85,11 @@ public String getVersion() {
             currentSensitivityAnalysisResult = runSystematicSensitivityAnalysisWithParametersSwitch(network, crac, computationManager, linearRaoParameters, useFallbackSensiParams);
         } catch (SensitivityComputationException e) { // Failure if some sensitivities are not computed
             resultVariantManager.deleteVariants(preOptimVariant, bestResultVariant);
-            return CompletableFuture.completedFuture(new RaoResult(RaoResult.Status.FAILURE));
+            LinearRaoResult resultExtension = new LinearRaoResult();
+            resultExtension.setSystematicSensitivityAnalysisParameters(useFallbackSensiParams.get());
```
Comment (baptsegu) [inline ra-optimisation/linear-rao/src/main/java/com/farao_community/farao/linear_rao/LinearRao.java:89]: and here we could then return a SystematicSensitivityAnalysisParameters.FAILURE, to say neither of both (or one if there is no fallback) configs worked.

**Pull request merged**